### PR TITLE
Match route

### PR DESCRIPTION
**Pull request by antonio-gg-dev**
## 📚 Description

Add the match() method to the Router that allows subscribing a route to multiple HTTP methods simultaneously.

## 🔖 Changes

- I classified the tests into several groups, there are already many tests but I have been able to see that they could be classified into four groups according to the functionality being tested, and so I have done it, I hope you like it!
- Added the match method described in the description.
- Routes class have been greatly refactored, now there is much less code and methods are reused much more, I think now it is easier to understand how it works.

### CODE REVIEW

**Review by Chemaclass — APPROVED** [review]
Awesome work 💡👏

File: tests/Unit/Router/RouterMatchTest.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+    /**
+     * @dataProvider notMatchesMethodsProvider
+     */
+    public function test_match_does_not_matches_other_methods(string $testMethod, array $givenMethods): void
+    {
+        $_SERVER['REQUEST_URI'] = 'https://example.org/expected/uri';
+        $_SERVER['REQUEST_METHOD'] = $testMethod;
+
+        $this->expectOutputString('');
```
Comment (JesusValeraDev) [inline tests/Unit/Router/RouterMatchTest.php:158]: Should we throw an exception with a `404 error code`? @Tito-Kati @Chemaclass

File: tests/Unit/Router/RouterMatchTest.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            $routes->get('other/uri', FakeController::class, 'basicAction');
+        });
+    }
+
+    public function test_not_respond_when_the_method_does_not_matches(): void
+    {
+        $_SERVER['REQUEST_URI'] = 'https://example.org/expected/uri';
+        $_SERVER['REQUEST_METHOD'] = 'GET';
+
+        $this->expectOutputString('');
```
Comment (JesusValeraDev) [inline tests/Unit/Router/RouterMatchTest.php:46]: Same here?

File: tests/Unit/Router/RouterMatchTest.php (hunk 1 above)
Comment (Chemaclass) [reply tests/Unit/Router/RouterMatchTest.php:158]: Yeah! We currently don't have support for 404, so this is a nice follow up feature to implement ;)

**Pull request merged**